### PR TITLE
Api 2.0/ort ovep

### PR DESCRIPTION
**Pull request by preetha-intel**
### Description
Add support for OpenVINO with SessionOptions_AppendExecutionProvider API of ONNXRuntime.

### Motivation and Context
Create backward compatibility with CLI options along with addition of new runtime options.

### CODE REVIEW

File: cmake/deps.txt [hunk 1]
```diff
@@ -41,4 +41,4 @@ safeint;https://github.com/dcleblanc/SafeInt/archive/ff15c6ada150a5018c5ef217240
 tensorboard;https://github.com/tensorflow/tensorboard/archive/373eb09e4c5d2b3cc2493f0949dc4be6b6a45e81.zip;67b833913605a4f3f499894ab11528a702c2b381
 cutlass;https://github.com/NVIDIA/cutlass/archive/refs/tags/v3.0.0.zip;0f95b3c1fc1bd1175c4a90b2c9e39074d1bccefd
 extensions;https://github.com/microsoft/onnxruntime-extensions/archive/94142d8391c9791ec71c38336436319a2d4ac7a0.zip;4365ac5140338b4cb75a39944a4be276e3829b3c
-eigen;https://gitlab.com/libeigen/eigen/-/archive/3.4/eigen-3.4.zip;ee201b07085203ea7bd8eb97cbcb31b07cfa3efb
\ No newline at end of file
+eigen;https://gitlab.com/libeigen/eigen/-/archive/3.4/eigen-3.4.zip;ee201b07085203ea7bd8eb97cbcb31b07cfa3efb
```
Comment (sfatimar) [inline cmake/deps.txt:44]: Is this change necessaey

File: docs/python/ReadMeOV.rst [hunk 2]
```diff
@@ -36,7 +36,7 @@ For more details on build and installation please refer to `Build <https://onnxr
 Usage
 ^^^^^
 
-By default, Intel® CPU is used to run inference. However, you can change the default option to either Intel® integrated or discrete GPU. 
+By default, Intel® CPU is used to run inference. However, you can change the default option to either Intel® integrated or discrete GPU.
```
Comment (sfatimar) [inline docs/python/ReadMeOV.rst:39]: Add integrated VPU here also

File: cmake/deps.txt (hunk 1 above)
Comment (preetha-intel) [reply cmake/deps.txt:44]: This change was added part of fixing merge conflicts. Not added externally.

File: onnxruntime/core/providers/openvino/openvino_provider_factory.cc [hunk 3]
```diff
@@ -63,12 +56,81 @@ struct OpenVINO_Provider : Provider {
...
+                                            //   with these values at runtime.
+    bool enable_vpu_fast_compile = false;   // [enable_vpu_fast_compile]: Fast-compile may be optionally enabled to
+                                            // speeds up the model's compilation to VPU device specific format.
+    const char* device_id = "";             // [device_id]: Selects a particular hardware device for inference.
+    size_t num_of_threads = 8;              // [num_of_threads]: Overrides the accelerator default value of number of
+                                            //  threads with this value at runtime.
+    const char* cache_dir = "";             // [cache_dir]: specify the path to
+                                            // dump and load the blobs for the model caching/kernel caching (GPU)
+                                            // feature. If blob files are already present, it will be directly loaded.
+    int num_streams = 1;                    // [num_streams]: Option that specifies the number of parallel inference
```
Comment (sfatimar) [inline onnxruntime/core/providers/openvino/openvino_provider_factory.cc:71]: Will adding num_stream
 lead to backward compatibility issue

Comment (preetha-intel) [reply onnxruntime/core/providers/openvino/openvino_provider_factory.cc:71]: A mapping is added for legacy API. 
By default its set to 1. 
> ov_options_converted_map["num_streams"] = "1"

File: onnxruntime/test/perftest/ort_test_session.cc [hunk 4]
```diff
@@ -458,72 +446,66 @@ OnnxRuntimeTestSession::OnnxRuntimeTestSession(Ort::Env& env, std::random_device
       if (key == "device_type") {
         std::set<std::string> ov_supported_device_types = {"CPU_FP32", "CPU_FP16", "GPU_FP32",
                                                            "GPU.0_FP32", "GPU.1_FP32", "GPU_FP16",
-                                                           "GPU.0_FP16", "GPU.1_FP16",
-                                                           "VPUX_FP16", "VPUX_U8"};
+                                                           "GPU.0_FP16", "GPU.1_FP16"};
```
Comment (sfatimar) [inline onnxruntime/test/perftest/ort_test_session.cc:449]: Why is VPUX Removed

File: onnxruntime/test/perftest/ort_test_session.cc [hunk 5]
```diff
@@ -458,72 +446,66 @@ OnnxRuntimeTestSession::OnnxRuntimeTestSession(Ort::Env& env, std::random_device
...
+          ov_options[key] = value;
         } else if (value.find("AUTO:") == 0) {
-          device_type = value;
+          ov_options[key] = value;
         } else {
           ORT_THROW(
               "[ERROR] [OpenVINO] You have selcted wrong configuration value for the key 'device_type'. "
               "Select from 'CPU_FP32', 'CPU_FP16', 'GPU_FP32', 'GPU.0_FP32', 'GPU.1_FP32', 'GPU_FP16', "
-              "'GPU.0_FP16', 'GPU.1_FP16', 'VPUX_FP16', 'VPUX_U8', or from"
+              "'GPU.0_FP16', 'GPU.1_FP16' or from"
```
Comment (sfatimar) [inline onnxruntime/test/perftest/ort_test_session.cc:462]: same here

File: onnxruntime/test/perftest/ort_test_session.cc [hunk 6]
```diff
@@ -458,72 +446,66 @@ OnnxRuntimeTestSession::OnnxRuntimeTestSession(Ort::Env& env, std::random_device
...
+          ov_options[key] = value;
         }
       } else if (key == "cache_dir") {
-        cache_dir = value;
-      } else {
-        ORT_THROW("[ERROR] [OpenVINO] wrong key type entered. Choose from the following runtime key options that are available for OpenVINO. ['device_type', 'device_id', 'enable_vpu_fast_compile', 'num_of_threads', 'cache_dir', 'enable_opencl_throttling|true'] \n");
+        ov_options[key] = value;
+      } else if (key == "num_streams") {
+        if (std::stoi(value) <= 0 && std::stoi(value) > 8) {
+          ORT_THROW("[ERROR] [OpenVINO] The value for the key 'num_streams' should be in the range of 1-8 \n");
```
Comment (sfatimar) [inline onnxruntime/test/perftest/ort_test_session.cc:500]: I think in future if we add more features then this error wont hold. We should have more generic error statement

**Comment by sfatimar** [comment]
LGTM

**Pull request merged**